cloud/aws-functions/orphaned_resources-openshift.py

```diff
@@ -0,0 +1,239 @@
+
+            logging.info(f"Removing security group with id: {security_group.id}")
+            try:
+                security_group.delete()
+            except ClientError as e:
+                logging.error(f"Deleting of security group failed with error: {e}")
+                continue
+    except Boto3Error as e:
+        logging.error(f"Deleting of security group failed with error: {e}")
+
+def wait_for_nat_gateway_delete(ec2, nat_gateway_id):
+    timeout = 300  # 5 min
+    attempt = 0
+    sleep_time = 10
+    attempts = timeout // sleep_time
+
+    while attempt < attempts:
+        try:
+            status = ec2.describe_nat_gateways(NatGatewayIds=[nat_gateway_id])['NatGateways'][0]['State']
+        except ec2.exceptions.ResourceNotFoundException:
+            logging.info(f"NAT gateway with id {nat_gateway_id} was not found.")
+            break
+
+        if status == 'deleted':
+            logging.info(f"NAT gateway with id {nat_gateway_id} was successfully deleted.")
+            break
+        logging.info(f"NAT gateway with id {nat_gateway_id} status is {status}. "
+                     f"Attempt {attempt}/{attempts}. Sleeping {sleep_time} seconds.")
+        sleep(sleep_time)
+        attempt += 1
+
+    else:
+        logging.error(f"NAT gateway with id {nat_gateway_id} was not deleted in {timeout} seconds.")
+
+def terminate_vpc(vpc_id, aws_region):
+    ec2_resource = boto3.resource('ec2', region_name=aws_region)
+
+    logging.info(f"Deleting load balancers for VPC {vpc_id}.")
+    delete_load_balancers(aws_region, vpc_id)
+
+    logging.info(f"Deleting NAT gateway for VPC {vpc_id}.")
+    delete_nat_gateway(aws_region, vpc_id)
+
+    logging.info(f"Deleting endpoints for VPC {vpc_id}.")
+    delete_vpc_ep(aws_region, vpc_id)
+    sleep(30)
+    logging.info(f"Deleting internet gateway for VPC {vpc_id}.")
+    delete_igw(ec2_resource, vpc_id)
+
+    logging.info(f"Deleting subnets for VPC {vpc_id}.")
+    delete_subnets(ec2_resource, vpc_id)
+
+    logging.info(f"Deleting route tables for VPC {vpc_id}.")
+    delete_route_tables(ec2_resource, vpc_id)
+
+    logging.info(f"Deleting security groups for VPC {vpc_id}.")
+
+    vpc_resource = ec2_resource.Vpc(vpc_id)
+    security_groups = vpc_resource.security_groups.all()
+    attempt = 0
+    while security_groups and attempt < 5:
+        delete_security_groups(security_groups)
+        security_groups = vpc_resource.security_groups.all()
+        attempt += 1
+
+    logging.info(f"Deleting VPC {vpc_id}.")
+    for attempt in range(0, 10):
+        logging.info(f"Deleting VPC {vpc_id}.. Attempt {attempt}/10")
+        try:
+            ec2_resource.Vpc(vpc_id).delete()
+        except ClientError as e:
+            logging.info(f"Failed to delete vpc, will try again. The error was: {e}. Sleeping 10 seconds")
+            sleep(10)
+            continue
+        break
+
+
+def lambda_handler(event, context):
+    aws_region = 'eu-west-3'
```

Comment:
What about other regions? PXC runs in eu-west-2? IIRC we have split jobs into different regions not to hit limits.

Reply:
I will create separate lambda for eu-west-2.